server/graphql/v2/query/TransactionGroupQuery.ts

```diff
@@ -0,0 +1,65 @@
+import { GraphQLNonNull, GraphQLString } from 'graphql';
+
+import { getFxRate } from '../../../lib/currency';
+import { sequelize, Transaction } from '../../../models';
+import { fetchAccountWithReference, GraphQLAccountReferenceInput } from '../input/AccountReferenceInput';
+import { GraphQLTransactionGroup } from '../object/TransactionGroup';
+
+import { getTransactionKindPriorityCase } from './collection/TransactionsCollectionQuery';
+
+const TransactionGroupQuery = {
+  type: GraphQLTransactionGroup,
+  description: '[!] Warning: this query is currently in beta and the API might change',
+  args: {
+    groupId: {
+      type: new GraphQLNonNull(GraphQLString),
+      description: 'The public id identifying the transaction group (ie: rvelja97-pkzqbgq7-bbzyx6wd-50o8n4rm)',
+    },
+    account: {
+      type: new GraphQLNonNull(GraphQLAccountReferenceInput),
+      description: 'Account associated to the transaction group',
+    },
+  },
+  async resolve(_, args) {
+    if (!args.account) {
+      throw new Error('You need to provide an account argument');
+    }
+    if (!args.groupId) {
+      throw new Error('You need to provide the groupId');
+    }
+    const account = await fetchAccountWithReference(args.account, { throwIfMissing: true });
+    const CollectiveId = account.id;
+
+    const transactions = await Transaction.findAll({
+      where: {
+        CollectiveId,
+        TransactionGroup: args.groupId,
+      },
+      order: [[sequelize.literal(getTransactionKindPriorityCase('Transaction')), 'ASC']],
+    });
+
+    const primaryTransaction = transactions[0]; // First transaction based on the ordering using getTransactionKindPriorityCase
+    const convertedAmounts = await Promise.all(
+      transactions.map(async t => {
+        const fxRate = await getFxRate(t.currency, account.currency, t.createdAt);
+        return Math.round(t.netAmountInCollectiveCurrency * fxRate);
```

Comment:
Loaders will make this faster

```suggestion
        const fxRate = await getFxRate(t.currency, account.currency, t.createdAt);
        return Math.round(t.netAmountInCollectiveCurrency * fxRate);
```

```suggestion
        return req.loaders.CurrencyExchangeRate.convert.load({
          amount: t.netAmountInCollectiveCurrency,
          fromCurrency: t.currency,
          toCurrency: account.currency
          date: t.createdAt
        });
```

Reply:
The loader does not have the date option

Reply:
You're right, the `convert` loaders doesn't support date. The `CurrencyExchangeRate.fxRate` one does though.
Not a change we need to do now, but surfacing that as a way to optimize the query later if needed.

Reply:
Ah, nice, will include it!